astropy/io/ascii/daophot.py

```diff
@@ -1,26 +1,99 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 """An extensible ASCII table reader and writer.
-
 daophot.py:
   Classes to read DAOphot table format
-
 :Copyright: Smithsonian Astrophysical Observatory (2011)
 :Author: Tom Aldcroft (aldcroft@head.cfa.harvard.edu)
 """
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import re
 import numpy as np
+import itertools as itt
+from collections import defaultdict
+
 from . import core
 from . import basic
 from . import fixedwidth
 from ...utils import OrderedDict
+from ...extern.six.moves import zip, map
+from .misc import first_true_index, first_false_index, groupmore
 
 
 class DaophotHeader(core.BaseHeader):
-    comment = r'\s*#K'
-    aperture_values = ''
+    """Read the header from a file produced by the IRAF DAOphot routine."""
+    def __init__(self):
+        core.BaseHeader.__init__(self)
+        self.comment = r'\s*#K'
```

Comment:
There is two ways to set class attributes like `comment`, `re_format`, `re_header_keyword` etc. Your method or setting these in the `__init__` works and that's what we initially did for all classes in io.ascii. However, we found that that makes is more cumbersome to derive something else from this class, so instead we decided that is is preferable to set them (as far as possible) like this:
```
class SomeHeader(core.BaseHeader):
    comment =''
```
See #2812 for more details. In this case it does not matter much because DAOPhot is such a specific format that it's very unlikely that we will derive many other classes from it, but I thought I'll just point that out. Change it or leave it as you wish.